package.json

```
@@ -1,22 +1,50 @@
 {
-  "name": "post-message-stream",
+  "name": "@metamask/post-message-stream",
   "version": "4.0.0",
   "description": "Sets up a duplex object stream over window.postMessage",
-  "main": "index.js",
+  "publishConfig": {
+    "registry": "https://registry.npmjs.org/",
+    "access": "public"
+  },
+  "main": "dist/cjs/index.js",
+  "module": "dist/esm/index.js",
+  "types": "dist/index.d.ts",
+  "files": [
+    "dist/"
+  ],
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "yarn build && yarn testOnly",
+    "testOnly": "jest --runInBand",
+    "test:watch": "jest --watch",
+    "prepublishOnly": "npm run build",
+    "lint": "eslint --ext .ts,.js ./src",
+    "build": "yarn run build:clean && yarn run build:compile && yarn run build:bundle",
+    "build:clean": "rm -rf ./dist && mkdir -p ./dist/esm && mkdir -p ./dist/cjs",
```

As an aside, I don't believe these paths need to exist, so we can simplify this script entry by only having the `rm` call:

```suggestion
    "build:clean": "rm -rf ./dist",
```

---

verified :+1: